src/plugin.js

```diff
@@ -184,31 +185,41 @@ const initPlugin = function(player, options) {
     if (display.getChild('closeButton')) {
       display.removeChild('closeButton');
     }
-    // Make the error display closeable, and we should get a close button
-    display.closeable(true);
+
     content.className = 'vjs-errors-dialog';
     content.id = 'vjs-errors-dialog';
     content.innerHTML =
-      `<div class="vjs-errors-content-container">
-        <h2 class="vjs-errors-headline">${this.localize(error.headline)}</h2>
-          <div><b>${this.localize('Error Code')}</b>: ${(error.type || error.code)}</div>
-          ${details}
-        </div>
-        <div class="vjs-errors-ok-button-container">
+     `<div class="vjs-errors-content-container">
+      <h2 class="vjs-errors-headline">${this.localize(error.headline)}</h2>
+        <div><b>${this.localize('Error Code')}</b>: ${(error.type || error.code)}</div>
+        ${details}
+      </div>`;
+
+    // Make the error display closeable, and we should get a close button
+    if (!('dismiss' in error) || error.dismiss) {
+      display.closeable(true);
+      content.innerHTML +=
+       `<div class="vjs-errors-ok-button-container">
           <button class="vjs-errors-ok-button">${this.localize('OK')}</button>
         </div>`;
+    }
+
     display.fillWith(content);
     // Get the close button inside the error display
     display.contentEl().firstChild.appendChild(display.getChild('closeButton').el());
     if (player.width() <= 600 || player.height() <= 250) {
       display.addClass('vjs-xs');
     }
 
+    let closeButton = display.el().querySelector('.vjs-close-button');
     let okButton = display.el().querySelector('.vjs-errors-ok-button');
-
-    videojs.on(okButton, 'click', function() {
+    let dismiss = function() {
       display.close();
-    });
+      player.error(null);
+    };
+
+    videojs.on(closeButton, 'click', dismiss);
```

Comment:
You shouldn't need to double-bind a handler for the `closeButton` or create a new listener for the `okButton`. Rather, you can add a listener to the `modalclose` event:

```js
display.on('modalclose', () => player.error(null));
```

This is particularly important to avoid leaving un-garbage-collected listeners around.